newsroom/search.py

```diff
@@ -528,9 +529,9 @@ def prefill_search_highlights(self, search, req):
                 },
             )
             elastic_highlight_query["fields"] = {
-                "body_html": {},
-                "headline": {},
-                "slugline": {},
+                "body_html": field_settings,
+                "headline": field_settings,
+                "slugline": field_settings,
```

Comment:
pls only configure fields which are selected in the `advanced_search["fields"]` config (or keep all if that config is empty)